src/vsphere_cpi/lib/cloud/vsphere/vm_creator.rb

```diff
@@ -114,6 +114,36 @@ def create(vm_config)
           raise "Unable to parse vmx options: 'vmx_options' is not a Hash"
         end
 
+        host = nil
+        # Before cloning we need to make sure the host is correctly picked up
+        # from cluster's host group if provided.
+        unless cluster.host_group.nil?
+          # placement_spec = VimSdk::Vim::Cluster::PlacementSpec.new
+          # placement_spec.config_spec = config_spec
+          # placement_spec.datastores = [datastore.mob]
+          # placement_spec.hosts = cluster.host_group_mob.host
+          # placement_result = cluster.mob.place_vm(placement_spec)
+          # require 'pry-byebug'
+          # binding.pry
+          # begin
+          #   host = placement_result.recommendations.first.action.first.target_host.name
+          # rescue => e
+          #   logger.warning("received #{e} while asking DRS for recommendation")
+          #   host = cluster.host_group_mob.host.find do |host|
+          #     host.runtime.connection_state == 'connected' &&
+          #       !host.runtime.in_maintenance_mode
+          #   end
+          # end
+          # @TA:TODO : Stop-Gap measure until DRS team solves bug for placeVM below.
+          # Selecting a random host out of all host groups to get initial placement correct
+          # Filtering it for healthy host.
+          host = cluster.host_group_mob.host.find do |host|
```

Comment:
if there is no healthy host, should you error out here?

Reply:
Done. Fixed. Good catch